src/main/java/hudson/plugins/ec2/SlaveTemplate.java

```diff
@@ -426,7 +426,7 @@ private EC2AbstractSlave provisionOndemand(TaskListener listener) throws AmazonC
                             updateRemoteTags(ec2, inst_tags, inst.getInstanceId());
                             break;
                         } catch (AmazonServiceException e) {
-                            if (e.getErrorCode().equals("InvalidInstanceRequestID.NotFound")) {
+                            if (e.getErrorCode().equals("InvalidInstanceID.NotFound")) {
```

Comment:
If you are unsure about previous ErrorCode or it was deprecated, then better do `if ( oneError || oneError) {` this will allow to have backward compatibility + put some comment that this error code doesn't exist anymore